crates/turborepo-auth/src/auth/login.rs

```diff
@@ -80,6 +80,25 @@ pub async fn login<T: Client + TokenClient + CacheClient>(
     let redirect_url = format!("http://{DEFAULT_HOST_NAME}:{DEFAULT_PORT}");
     let mut login_url = Url::parse(login_url_configuration)?;
 
+    // Construct the success redirect URL from the base login URL.
+    login_url
+        .path_segments_mut()
+        .map_err(|_: ()| Error::LoginUrlCannotBeABase {
+            value: login_url_configuration.to_string(),
+        })?
+        .extend(["turborepo", "success"]);
+
+    let success_url = Url::parse(login_url.as_str())?;
+
+    // Undo the success redirect URL.
+    login_url
+        .path_segments_mut()
+        .map_err(|_: ()| Error::LoginUrlCannotBeABase {
+            value: login_url_configuration.to_string(),
+        })?
+        .clear();
```

Comment:
Can you clone it instead of adding in path segments and then removing them? This feels very weird, even for rust

Reply:
yeah this was my quick
> get something out before EOD
so brain machine wasn't fully there 😂